### PR TITLE
Add lock_stronghold and destroy methods to Python binding

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add lock_stronghold and destroy methods to Python binding

## Links to any relevant issues

Fixes https://github.com/iotaledger/wallet.rs/issues/1733

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Added example

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/python/examples/destroy.py
Comment (kwek20) [inline bindings/python/examples/destroy.py:1]: I understand we have this for testing the deletion, but is keeping it an examplereally needed?

Comment (Thoralf-M) [reply bindings/python/examples/destroy.py:1]: What speaks against having an example for this? I think it's better to have it than not

Comment (kwek20) [reply bindings/python/examples/destroy.py:1]: it feels very specific to the bindings glue. if its something that should always be done it should be automated. This example doesnt actually directly use the methods we added either. 

And if were actually testing account getting etc, we should do that in the main crate

Comment (Thoralf-M) [reply bindings/python/examples/destroy.py:1]: It uses `.destroy()` and recreates the account manager, the case that didn't work in the issue, but I agree that it's not really general shown and also specific to use cases where you need to recreate it
Should I change it so it just creates the account manager, destroys it and then recreates it, without anything else?

File: bindings/python/native/tests/test_account_manager.py [hunk 1]
```diff
@@ -195,32 +195,32 @@ def test_account_manager_stop_background_sync():
...
-    except ValueError as e:
-        assert 'Input value is < dust protection value' in str(e)
-    else:
-        try:
-            manager.send_migration_bundle(
-                [legacy_node], bundle['bundle_hash'], min_weight_magnitude)
-
-        except ValueError as e:
-            assert 'input list is empty' in str(e)
+# def test_get_migration_data():
```
Comment (kwek20) [inline bindings/python/native/tests/test_account_manager.py]: We want this test entirely commented out? Sort of what git history is for :)

Comment (Thoralf-M) [reply bindings/python/native/tests/test_account_manager.py]: Will remove, this stuff should never be touched anymore anyways

**Review by kwek20 — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**